yaml/converter/legacy/internal/config.go

```diff
@@ -21,6 +22,7 @@ type Config struct {
 	Pipeline Containers
 	Services Containers
 	Branches Constraint
+	Matrix   Matrix
```

Comment:
lets change this to `Matrix interface{}` and then delete the `matrix.go` file since it is otherwise empty.